src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1496,32 +1521,36 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
         case Some(_) => DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Conflict, s"Workspace ${workspaceRequest.namespace}/${workspaceRequest.name} already exists")))
         case None =>
           val workspaceId = UUID.randomUUID.toString
-          DBIO.from(gcsDAO.setupWorkspace(userInfo, workspaceRequest.namespace, workspaceId, workspaceRequest.toWorkspaceName, workspaceRequest.realm)) flatMap { googleWorkspaceInfo =>
-            val currentDate = DateTime.now
-
-            val accessGroups = googleWorkspaceInfo.accessGroupsByLevel.map { case (a, g) => (a -> RawlsGroup.toRef(g))}
-            val intersectionGroups = googleWorkspaceInfo.intersectionGroupsByLevel map { _.map { case (a, g) => (a -> RawlsGroup.toRef(g))} }
-
-            val workspace = Workspace(
-              namespace = workspaceRequest.namespace,
-              name = workspaceRequest.name,
-              realm = workspaceRequest.realm,
-              workspaceId = workspaceId,
-              bucketName = googleWorkspaceInfo.bucketName,
-              createdDate = currentDate,
-              lastModified = currentDate,
-              createdBy = userInfo.userEmail,
-              attributes = workspaceRequest.attributes,
-              accessLevels = accessGroups,
-              realmACLs = intersectionGroups getOrElse accessGroups
-            )
-
-            val groupInserts =
-              googleWorkspaceInfo.accessGroupsByLevel.values.map(dataAccess.rawlsGroupQuery.save) ++
-              googleWorkspaceInfo.intersectionGroupsByLevel.map(_.values.map(dataAccess.rawlsGroupQuery.save)).getOrElse(Seq.empty)
-
-            DBIO.seq(groupInserts.toSeq:_*) andThen
-              dataAccess.workspaceQuery.save(workspace).flatMap(ws => op(SlickWorkspaceContext(ws)))
+          dataAccess.rawlsBillingProjectQuery.load(RawlsBillingProjectName(workspaceRequest.namespace)) flatMap { project =>
+            DBIO.from(gcsDAO.setupWorkspace(userInfo, project.get, workspaceId, workspaceRequest.toWorkspaceName, workspaceRequest.realm)) flatMap { googleWorkspaceInfo =>
```

Comment:
This one is too hard for the moment. Besides whitespace, we did actually touch this code so it is not actually worse.